docker/compute_worker/compute_worker.py

```diff
@@ -607,7 +616,20 @@ def start(self):
             self.watch_file(),
             loop=loop,
         )
-        loop.run_until_complete(gathered_tasks)
+        signal.signal(signal.SIGALRM, alarm_handler)
+        signal.alarm(self.execution_time_limit)
+        try:
+            loop.run_until_complete(gathered_tasks)
+        except ExecutionTimeLimitExceeded:
```

Comment:
May make sense to do file uploading here instead of on exception. Like put it in a finally clause? That way we don't have to set the key to None as well